testplan/web_ui/testing/src/AssertionPane/AssertionSummary.js

```diff
@@ -0,0 +1,104 @@
+import React, {Component, Fragment} from 'react';
+import PropTypes from 'prop-types';
+import {Col, Row} from 'reactstrap';
+import AssertionGroup from './AssertionGroup';
+/**
+ * Component used to render a large test case when summarize = true
+ */
+
+class SummaryBaseAssertion extends Component  {
+    displayAssertionGroup(assertion_entries){
+        let group_assertion_jsx = [];
```

Comment:
As an exercise you might want to think how you could transform this loop-based approach to a more functional approach using assertion_entries.map(...). Not required, just if you are curious :)

Reply:
When I was implementing this i looked up the best practice for looping through arrays, and i was surprised that this was the recommended way (and i was slightly sad :[ ) i originally did the 'for(in)' approach but found out it wasn't recommended (python has converted me) D: i think almost every implementation looks cleaner than this :')

https://stackoverflow.com/questions/6208964/how-to-loop-over-an-array-with-javascript

Reply:
I think the answer to this SO is more up-to-date for modern JS: https://stackoverflow.com/questions/3010840/loop-through-an-array-in-javascript

In summary: `for ... in` is bad. Arrays have a `.forEach()` method you can use for iteration, though `.map()` is the recommended way to build a new array from an existing one (which is what you are doing here).